src/Stateless/StateMachine.cs

```diff
@@ -33,21 +44,25 @@ class QueuedTrigger
         /// </summary>
         /// <param name="stateAccessor">A function that will be called to read the current state value.</param>
         /// <param name="stateMutator">An action that will be called to write new state values.</param>
-        public StateMachine(Func<TState> stateAccessor, Action<TState> stateMutator) : this()
+        /// <param name="firingMode">Optional specification of fireing mode.</param>
+        public StateMachine(Func<TState> stateAccessor, Action<TState> stateMutator, FiringMode firingMode = FiringMode.Queued) : this()
         {
             _stateAccessor = stateAccessor ?? throw new ArgumentNullException(nameof(stateAccessor));
             _stateMutator = stateMutator ?? throw new ArgumentNullException(nameof(stateMutator));
+            _firingMode = firingMode;
         }
 
         /// <summary>
         /// Construct a state machine.
         /// </summary>
         /// <param name="initialState">The initial state.</param>
-        public StateMachine(TState initialState) : this()
+        /// <param name="firingMode">Optional specification of fireing mode.</param>
+        public StateMachine(TState initialState, FiringMode firingMode = FiringMode.Queued) : this()
         {
             var reference = new StateReference { State = initialState };
             _stateAccessor = () => reference.State;
             _stateMutator = s => reference.State = s;
+            _firingMode = firingMode;
```

Comment:
If we check that the firing mode is either `Immediate` or `Queued` here, we can get rid of the switch and default case below? (Just a thought - may not really be a big advantage).

Reply:
It is certainly possible to add an Action<TTrigger, object[]> FireHandler that is configured in the constructor.  InternalFire is then reduced to 

        void InternalFire(TTrigger trigger, params object[] args)
        {
            FireHandler(trigger, args);
        }

It would shave off a few CPU cycles...